qiskit/utils/run_circuits.py

```diff
@@ -197,13 +198,14 @@ def _safe_submit_qobj(
             logger.warning(
                 "FAILURE: Can not get job id, Resubmit the qobj to get job id." "Error: %s ", ex
             )
+    else:
+        raise QiskitError("Can not submit qobj correctly. Please resubmit the qobj.")
```

Comment:
I think the error raised here should indicate the max job retries limit was reached.

In general I will note that this code is used by algorithms not by end users. The algorithm will fail and the error msg that seems to indicate resubmitting job will be meaningless in that context since this circuit execution logic is buried within the algorithm and the QuantumInstance was designed to handle retries etc so it did not need to be replicated throughout the algorithms but was in one central place.

Reply:
Okay, I shall change that
Just a small query, this error message needs to be changed pertaining to every added `Exception` right?

Reply:
The message ideally should be meaningful to the context where it was raised so if the rest accurately report what happened in regards of the error they would be fine. In the above since the logic has retried a bunch of times already itself and failed to submit it, it seems unlikely doing it again, as the message suggests is going to work. If there is another similar place then yes it would think it should be changed there too.